app/models/foreman_tasks/task/dynflow_task.rb

```diff
@@ -15,7 +15,8 @@ def update_from_dynflow(data)
       self.start_at       = string_to_time(utc_zone, data[:start_at]) if data[:start_at]
       self.start_before   = string_to_time(utc_zone, data[:start_before]) if data[:start_before]
       self.parent_task_id ||= begin
-                                if main_action.present? && main_action.caller_execution_plan_id
+                                if main_action.present? && main_action.respond_to?(:caller_execution_plan_id) &&
```

Comment:
Nice catch: I actually didn't know you can use `try` against non-nil objects in case they're not defining methods. I went even a bit further, and eliminated it just to `main_action.try(:caller_execution_plan_id)`, as it also implies that the main_action would be present.